src/Microsoft.Azure.WebJobs.Extensions.EventHubs/EventDataEx.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.EventHubs;
+
+namespace Microsoft.Azure.WebJobs.EventHubs
+{
+    /// <summary>
+    /// Extended <see cref="EventData"/> class allowing additional metadata to be
+    /// specified.
+    /// </summary>
+    public class EventDataEx : EventData
+    {
+        public EventDataEx(byte[] array) : base(array) { }
+
+        public EventDataEx(ArraySegment<byte> arraySegment) : base(arraySegment) { }
+
+        /// <summary>
+        /// Gets or sets the partition key that should be used when the
+        /// event is sent.
+        /// </summary>
+        public string PartitionKey { get; set; }
```

Comment:
We could make this immutable, however the base EventData type exposes a Properties dictionary that can have values added/removed at any time. So I lean towards having a friendlier programming model for this, rather than having ctor optional params for every extended property.